samples/jenkins/README.md

```diff
@@ -1,5 +1,373 @@
+
+  ![production-job-2.png](img/production-job-2.png)
+
+## Scenario implementation
+The scenario implementation is split into three Scripted Jenkins pipelines. The definition of each pipeline is written
+into text files, located in the [pipelines](pipelines) directory. The division matches the CI/CD pipelines described in the base scenario:
+- development
+- test
+- production
+
+All of the pipelines have some common step, to set up a working environment, install dependencies or to define the stage:
+1. `node` - allocates an executor on a build agent (node) and runs further code in the context of a workspace on that agent. 
+In this sample, the default agent is used.
+2. `withCredentials([file(credentialsId: 'CPDCTL_CPD402_CONFIG', variable: 'CPDCONFIG')])` - loads `cpdctl` configuration file
+from a secret file uploaded to Jenkins in the Credentials section.
+3. `checkout scm` - the code from the Source Control Management defined in the job configuration.
+4. `stage` - defines the pipeline stage
+5. `sh 'bash -e ...'` - executes bash script, waits for the result and fails the pipeline when script returns non-zero exit code
+
+### Development pipeline
+The pipeline, that trains the model, promotes it and export assets as a Jenkins artifact consists of stages:
+1. Install cdpctl - downloads and installs the latest `cdpctl` version using [install_cpdctl.sh](scripts/helpers/install_cpdctl.sh) script
+2. Train model - trains the model using notebook and datasets located in Watson Studio with the [train_model.sh](scripts/dev/train_model.sh) script 
+3. Promote to DEV - executes [promote_to_dev_space.sh](scripts/dev/promote_to_dev_space.sh) to promote trained model from the 
+project to the development deployment space
+4. Export assets - executes [export_dev_space.sh](scripts/dev/export_dev_space.sh) to export assets from the development deployment space to an archive, 
+then, using `archiveArtifacts` step, attach the archive to the Jenkins build as a build artifacts.
+5. Cleanup - executes [cleanup.sh](scripts/dev/cleanup.sh) to delete all models, assets and jobs from the project.
+
+### Test pipeline
+Import of the artifacts from the previous job, model deployment and evaluation is defined in the test pipeline with the following stages:
+1. Install cdpctl - same as in the development pipeline 
+2. Import assets - using `copyArtifacts` step downloads artifacts archived in development pipeline, then imports the archive to 
+the newly created deployment space with [create_test_space.sh](scripts/test/create_test_space.sh) script.
+3. Deploy model - deploys model with [create_model_batch_deployment.sh](scripts/test/create_model_batch_deployment.sh) script.
+4. Evaluate model - evaluates model deployment using [create_script_batch_deployment.sh](scripts/test/create_script_batch_deployment.sh) script.
+5. Cleanup - executes [cleanup.sh](scripts/test/cleanup.sh) to delete all models, assets and jobs from the project.
```

Comment:
The cleanup.sh in test_pipeline deletes the whole test-space with the assets, not from the project